internal/tools/cypher/json_tagged_types.go

```diff
@@ -0,0 +1,50 @@
+package cypher
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type Params map[string]any
```

Comment:
It would be nice to add godoc comments for exported types and functions so this is easier to understand:


```suggestion
// Params is a map of Cypher query parameters with custom JSON unmarshaling
// that preserves numeric types correctly for Neo4j.
//
// When unmarshaling from JSON:
//   - Whole numbers (e.g., 1, 42, -10) become int64
//   - Numbers with fractional parts (e.g., 1.5, 3.14) become float64
//   - Numbers with decimal notation but no fraction (e.g., 10.0) become float64
//   - Other types (strings, booleans, null) are preserved as-is
type Params map[string]any
```